src/NSwag.SwaggerGeneration.AspNetCore/OperationParameterProcessor.cs

```diff
@@ -0,0 +1,334 @@
+                                    else
+                                    {
+                                        // If we are not reading from the body, then treat this as a primitive.
+                                        // This may seem odd, but it allows for primitive -> custom complex-type bindings which are very common
+                                        // In this case, the API author should use a TypeMapper to define the parameter
+                                        await AddPrimitiveParametersFromUriAsync(
+                                            context, apiParameter, httpPath, parameterName, parameter, parameterInfo).ConfigureAwait(false);
+                                    }
+                                }
+                            }
+                            else
+                                await AddPrimitiveParameterAsync(parameterName, context.OperationDescription.Operation, parameter, context.SwaggerGenerator).ConfigureAwait(false);
+                        }
+                    }
+                }
+            }
+
+            if (_settings.AddMissingPathParameters)
+            {
+                foreach (Match match in Regex.Matches(httpPath, "{(.*?)(:(([^/]*)?))?}"))
+                {
+                    var parameterName = match.Groups[1].Value;
+                    if (context.OperationDescription.Operation.Parameters.All(p => !string.Equals(p.Name, parameterName, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        var parameterType = match.Groups.Count == 5 ? match.Groups[3].Value : "string";
+                        var operationParameter = context.SwaggerGenerator.CreateUntypedPathParameter(parameterName, parameterType);
+                        context.OperationDescription.Operation.Parameters.Add(operationParameter);
+                    }
+                }
+            }
+
+            RemoveUnusedPathParameters(context.OperationDescription, httpPath);
+            UpdateConsumedTypes(context.OperationDescription);
+
+            EnsureSingleBodyParameter(context.OperationDescription);
+
+            return true;
+        }
+
+        private void EnsureSingleBodyParameter(SwaggerOperationDescription operationDescription)
+        {
+            if (operationDescription.Operation.ActualParameters.Count(p => p.Kind == SwaggerParameterKind.Body) > 1)
+                throw new InvalidOperationException("The operation '" + operationDescription.Operation.OperationId + "' has more than one body parameter.");
+        }
+
+        private void UpdateConsumedTypes(SwaggerOperationDescription operationDescription)
+        {
+            if (operationDescription.Operation.ActualParameters.Any(p => p.Type == JsonObjectType.File))
+                operationDescription.Operation.Consumes = new List<string> { "multipart/form-data" };
+        }
```

Comment:
@pranavkm - do we do anything like this in MVC? We should think about adding an implicit `[Consumes(...)]` when you have a form file

Reply:
@RSuter  btw, our philosophy of such things is that we try to use ApiExplorer to describe the behavior of the framework in sensible situations.

So understanding that `IFormFile` and `IFormFileCollection` -> file uploads is something we should do. It's also fine if NSwag does it as well, but we want to patch holes like this in our story.

Reply:
Yeah, that seems fine. We'll add inference as part of https://github.com/aspnet/Mvc/issues/6847 for 2.1.0 and leave this piece of code around for legacy support